src/openvtb/components/taken/api/serializers.py

```diff
@@ -0,0 +1,192 @@
+        },
+        group_field="details",
+        same_model=False,
+    )
+    discriminator_field = "taak_soort"
+
+    class Meta:
+        model = ExterneTaak
+        fields = (
+            "url",
+            "uuid",
+            "titel",
+            "status",
+            "startdatum",
+            "handelings_perspectief",
+            "einddatum_handelings_termijn",
+            "datum_herinnering",
+            "toelichting",
+            "taak_soort",
+            "details",
+        )
+        validators = [
+            StartBeforeEndValidator("startdatum", "einddatum_handelings_termijn"),
+        ]
+        extra_kwargs = {
+            "uuid": {"read_only": True},
+            "details": {"required": True},
+            "taak_soort": {
+                "required": True,
+                "allow_null": False,
+            },
+            "url": {
+                "view_name": "taken:externetaak-detail",
+                "lookup_field": "uuid",
+                "help_text": _("De unieke URL van de externe taak deze API."),
+            },
+        }
+
+    def _init_taak_soort(self, taak_soort, partial=False):
+        """
+        Initialize `taak_soort`:
+
+        - If `taak_soort` is already set:
+            - if it also appears in `initial_data`, raise a ValidationError
+            - Otherwise, insert it into `initial_data` as default
+
+        - If `partial=True` and is not passed to the initial data:
+            - Add `taak_soort` value into `initial_data`
+        """
+
+        initial_data = getattr(self, "initial_data", None)
+        instance = getattr(self, "instance", None)
+
+        if initial_data is None:
+            return
+
+        if taak_soort:
+            if self.discriminator_field in initial_data:
+                raise serializers.ValidationError(
+                    {
+                        self.discriminator_field: _(
+                            "Dit veld wordt automatisch ingevuld; het kan niet worden geselecteerd."
+                        )
+                    },
+                    code="invalid",
+                )
+            initial_data[self.discriminator_field] = taak_soort
+            return
+
+        if partial and instance and self.discriminator_field not in initial_data:
+            initial_data[self.discriminator_field] = instance.taak_soort
+            return
+
+    def __init__(self, *args, **kwargs):
+        partial = kwargs.get("partial", False)
+        # partial pass for PATCH the content of the details
+        for discriminator in self.discriminator.mapping.values():
+            discriminator.partial = partial
+
+        super().__init__(*args, **kwargs)
+        if context := kwargs.get("context", {}):
+            taak_soort = context.get(self.discriminator_field, None)
+            self._init_taak_soort(taak_soort, partial)
+
+    def validate(self, attrs):
+        renderer = CamelCaseJSONRenderer()
```

Comment:
Will approve but I think if this is the default renderer in `REST_FRAMEWORK` settings, this here might not be needed?